sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,299 @@
+-- Query first observations for Firefox Android Clients.
+WITH first_seen AS (
+  SELECT
+    client_id,
+    first_seen_date,
+    country AS first_reported_country,
+    isp AS first_reported_isp,
+    DATETIME(first_run_date) AS first_run_datetime,
+    normalized_channel AS channel,
+    device_manufacturer,
+    device_model,
+    normalized_os_version AS os_version
+  FROM
+    `mozdata.fenix.baseline_clients_first_seen`
+  WHERE
+    submission_date = @submission_date
+    AND normalized_channel = 'release'
+),
+-- Find earliest data per client from the first_session ping.
+first_session_ping AS (
+  SELECT
+    client_info.client_id AS client_id,
+    MIN(SAFE.PARSE_DATETIME('%F', SUBSTR(client_info.first_run_date, 1, 10))) AS first_run_datetime,
+    ARRAY_AGG(
+      metrics.string.first_session_campaign
+      ORDER BY
+        metrics.datetime.first_session_timestamp ASC
```

Comment:
This will be the same for every install, so if we see two of these from the same client, the order will be nondeterministic. Let's stick with `submission_timestamp` for ordering.

Reply:
Good point, thanks.